pkg/edition/java/lite/forward.go

```diff
@@ -58,28 +62,37 @@ func Forward(
 		return
 	}
 
-	log.Info("forwarding connection", "backendAddr", netutil.Host(dst.RemoteAddr()))
+	// add connection to the connection counter
+	if route.Strategy == config.StrategyLeastConnections {
```

Comment:
This has a race condition. You should use an atomic increase not a load + store. This code runs in parallel to every connection.